### PR TITLE
Add zb score threshold variable ##signatures

### PR DESCRIPTION
**Pull request by swoops**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [x] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed) 
Book update [pending here](https://github.com/radareorg/radare2book/pull/271/commits)

**Detailed description**

Small change to finally make use of the score threshold variable in zb commands. Setting `e zign.bestthresh = 0.5` will limit requires results to have a similarity score above `0.5`.  Setting a good threshold value should increase the speed of the algorithm by reducing the number of times the Levenshtein Distance is executed.

I am not a big fan of `bestthresh`, so if you have a better name I am happy to change it.  

**Test plan**

Tests were submitted to take care of common edge cases. The bulk of this feature was previously implemented, I just added the variable.

**Closing issues**

...

### CODE REVIEW

File: libr/core/cconfig.c [hunk 1]
```diff
@@ -3261,6 +3261,7 @@ R_API int r_core_config_init(RCore *core) {
 	SETBPREF ("zign.autoload", "false", "Autoload all zignatures located in " R_JOIN_2_PATHS ("~", R2_HOME_ZIGNS));
 	SETPREF ("zign.diff.bthresh", "1.0", "Threshold for diffing zign bytes [0, 1] (see zc?)");
 	SETPREF ("zign.diff.gthresh", "1.0", "Threshold for diffing zign graphs [0, 1] (see zc?)");
+	SETPREF ("zign.bestthresh", "0.0", "Threshold for zb commands");
```
Comment (trufae) [inline libr/core/cconfig.c:3264]: I dont like the var name or the description. But it seems like an useful feature.

Proposals:

- zign.filter
- use zign.threshold. As prefix for diff.b,g,min..
- find better name instead of threshold

For the dexcription i would say zign.diff.* are also poor descriptions. Maybe better to say like “ filter fir matches having minimum threshold for bytes matching”

Comment (XVilka) [reply libr/core/cconfig.c:3264]: I would say just `zign.threshold` with a better description than just "Threshold for zb commands".

Comment (trufae) [reply libr/core/cconfig.c:3264]: make clear the idfference with `	SETPREF ("zign.diff.bthresh", "1.0", "Threshold for diffing zign bytes [0, 1] (see zc?)");
`

File: libr/core/cmd_zign.c [hunk 2]
```diff
@@ -1035,6 +1035,16 @@ static RSignItem *item_frm_signame(RAnal *a, const char *signame) {
 	return it;
 }
 
+double get_zb_threshold(RCore *core) {
```
Comment (trufae) [inline libr/core/cmd_zign.c:1038]: static

**Review by trufae — CHANGES_REQUESTED** [review]

**Comment by swoops** [comment]
Sorry was away for a bit, I was busy being anxious over r2con :)

I made an update, let me know if you like it. I am pretty bad at naming and describing things so feel free to make more suggestions.

**Review by trufae — APPROVED** [review]
Looks good to me now! :)
